### ISSUE TITLE
enforce "important-dates" deadlines in UTC

### ISSUE THREAD

**Issue by jennifer-richards** [ietf-tools/datatracker#4436]
### Describe the issue

Many items on the list at https://datatracker.ietf.org/meeting/important-dates/ indicate that their deadline is in UTC. While reworking the timezone handling, verify whether these are implemented to match and fix if not.

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by bomturbo** [comment]
> ### Describe the issue
> Many items on the list at https://datatracker.ietf.org/meeting/important-dates/ indicate that their deadline is in UTC. While reworking the timezone handling, verify whether these are implemented to match and fix if not.
> 
> ### Code of Conduct
> * [x]  I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by jennifer-richards** [comment]
Going through the list of dates for 115, I see:
* IETF registration opens
* WG/BOF scheduling begins
* Cut-off for BOF requests
* Early bird reg cutoff (in UTC)
* Cut-off for WG session requests (in UTC)
* Cut-off for ADs to approve BOFs (in UTC)
* Prelim agenda published
* Cut-off for WG/BOF rescheduling requests (in UTC)
* Final agenda published
* I-D submission cut-off (in UTC)
* Standard reg cut-off (in UTC)
* Draft WG agendas due (in UTC)
* Reg cancellation cut-off (in UTC)
* Revised WG agendas due (in UTC)
* Proceedings submission cut-off (in UTC)
* Proceedings corrections cut-off (in UTC)
Of the dates that claim to be UTC, I believe the registration-related ones would be enforced outside the Datatracker. This leaves
* Cut-off for WG session requests (in UTC)
* Cut-off for ADs to approve BOFs (in UTC)
* Cut-off for WG/BOF rescheduling requests (in UTC)
* I-D submission cut-off (in UTC)
* Draft WG agendas due (in UTC)
* Revised WG agendas due (in UTC)
* Proceedings submission cut-off (in UTC)
* Proceedings corrections cut-off (in UTC)
Need to check which of these are enforced in the Datatracker.

**Comment by jennifer-richards** [comment]
It looks like:
* WG session requests are locked by setting the `meeting.session_request_lock_message`, so it's up to the person setting that value to do so in UTC
* BOF approval is done out-of-band from the datatracker, so again up to the responsible people to use UTC
* I-D submission cut-off was already done in UTC. The code around this is cleaned up a bit in #4612 but it was already doing the right thing
* agendas and proceedings, at least on the main materials page, were being checked against the server date so need to be changed to use UTC. May need further attention in the `session_details` view as well

tl;dr I think only the agenda and proceedings cutoffs need correction within the datatracker

**Comment by rjsparks** [comment]
Those should be fixed, but the priority isn't screaming high as the error isn't horrible - the secretariat can easily deal with exceptions in the 8-ish hours that are not yet right.

**Comment by jennifer-richards** [comment]
Ok - I already have a fix that's just a couple lines. The previous behavior on feat/tzaware was not self-consistent: the meeting materials page would cut off according to server time, but the edit materials page operated in meeting time. The patch makes both UTC.

**Comment by jennifer-richards** [comment]
Fixed on feat/tzaware branch, so closing.
